src/main/Core/ImageGenerator/Linux/LinuxAppIconExtractor.ts

```diff
@@ -271,15 +273,52 @@ export class LinuxAppIconExtractor implements FileIconExtractor {
     }
 
     private async getIconThemeName(): Promise<string> {
-        const currentDesktopEnvironment = this.environmentVariablePRovider.get("XDG_SESSION_DESKTOP");
-
-        const iconThemeNameMap: Record<"cinnamon" | "gnome", () => Promise<string>> = {
-            cinnamon: () =>
-                this.commandlineUtility.executeCommand("gsettings get org.cinnamon.desktop.interface icon-theme"),
-            gnome: () => this.commandlineUtility.executeCommand("gsettings get org.gnome.desktop.interface icon-theme"),
+        const iconThemeNameMap: Record<LinuxDesktopEnvironment, () => Promise<string>> = {
+            GNOME: (): Promise<string> => {
```

Comment:
I think these type declarations are not necessary:

```suggestion
            GNOME: () => {
```

Also, you can simplify this by removing the redudant return statement:

```typescript
GNOME: () => this.commandlineUtility.executeCommand("gsettings get org.gnome.desktop.interface icon-theme")
```